pdns/pdnsutil.cc

```diff
@@ -1001,15 +1001,19 @@ int editZone(DNSSECKeeper& dk, const DNSName &zone) {
  editMore:;
   struct stat statbefore, statafter;
   stat(tmpnam,&statbefore);
-  cmdline=editor+" ";
+  cmdline=" ";
   if(gotoline > 0)
     cmdline+="+"+std::to_string(gotoline)+" ";
   cmdline += tmpnam;
-  int err=system(cmdline.c_str());
+  int err=system((editor + cmdline).c_str());
   if(err) {
-    unixDie("Editing file with: '"+cmdline+"', perhaps set EDITOR variable");
+    err=system(("vi" + cmdline).c_str());
```

Comment:
Silently falling back to vi when the editor exits with a failure reason does not make sense.